### PR TITLE
feat: Added shadcn ui with ui installer

### PR DESCRIPTION
**Pull request by kythonlk**
I added shadcn/ui with shadcn component install cli option helper

### CODE REVIEW

File: logic/helpers/shadcn.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * @throws {Error} If the component name is not found in the `componentsList`.
+ * @throws {Error} If the shell command execution fails.
+ * 
+ * @description
+ * This function checks if the provided component exists in the predefined `componentsList`.
+ * If found, it constructs a command using the `npx shadcn@latest add <component>` syntax and executes it.
+ * If the component is not found or there is an error during execution, it logs an error to the console.
+ */
+
+export function runComponentCommand(component) {
```
Comment (x5engine) [inline logic/helpers/shadcn.js:31]: This needs reading please or the coverage will be low

File: logic/componentList.js [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 
-    // Bootstrap Components
-    'Button', 'Dropdown', 'Form', 'InputGroup', 'Card', 'Table',
+  // Bootstrap Components
+  'Button', 'Dropdown', 'Form', 'InputGroup', 'Card', 'Table',
 
-    // Ant Design Components
-    'Button', 'Select', 'Input', 'Grid', 'Card', 'Table', 'Modal',
+  // Ant Design Components
+  'Button', 'Select', 'Input', 'Grid', 'Card', 'Table', 'Modal',
```
Comment (x5engine) [inline logic/componentList.js:9]: Let's add eslint and lint pre commit so we have the same formatting please 
Thank you

**Review by x5engine — CHANGES_REQUESTED** [review]

**Comment by x5engine** [comment]
Amazing pr thank you

Just do these two small change I'll merge it

File: logic/helpers/shadcn.js (hunk 1 above)
Comment (kythonlk) [reply logic/helpers/shadcn.js:31]: I didn't get you. Can you please elaborate

File: logic/componentList.js (hunk 2 above)
Comment (kythonlk) [reply logic/componentList.js:9]: Sure thing

File: logic/helpers/shadcn.js (hunk 1 above)
Comment (x5engine) [reply logic/helpers/shadcn.js:31]: Can you some test cases for this function only?

That's about it but amazing work

**Pull request merged**